### PR TITLE
Support releasing invisibility spells

### PR DESCRIPTION
**Pull request by jandersson**
This releases invisibility spells in a very similar fashion to the cyclic release. Addresses #945 
The idea is to use the common method `release_invisibility` before doing an action which would fail on the grounds of being invisible

### CODE REVIEW

File: common.lic [hunk 1]
```diff
@@ -102,7 +104,10 @@ module DRC
     result = DRC.bput("rummage /#{parameter} my #{container}", 'but there is nothing in there like that\.', 'looking for .* and see .*', 'While it\'s closed', 'I don\'t know what you are referring to', 'You feel about')
 
     case result
-    when 'but there is nothing in there like that.', 'While it\'s closed', 'I don\'t know what you are referring to', 'You feel about'
+    when 'You feel about'
+      release_invisibility
```
Comment (rpherbig) [inline common.lic]: @rcuhljr Will something weird happen with common and common-arcana both `custom_require`ing each other?

**Comment by jandersson** [comment]
I'm looking at line 60 in hunting-buddy that was suggested to be changed. What does this do? It looks like it releases invisibility after hunting

**Comment by rpherbig** [comment]
That line was added to fix #801. There seem to be two general approaches:
1. Call `release_invisibility` in each place that it could cause problems (I think you'd need to test `sell-loot` and `crossing-repair`)
2. Call `release_invisibility` in hunting-buddy or training-manager when we transition "states"

I'm in favor of the first option, because it is safer to call those scripts manually.

**Comment by jandersson** [comment]
I think the first option makes the constraints clearer (ie. have to be visible for this action to succeed) but the second one is easier. I like the first one because it lets me run around invisible for longer :]

**Comment by jandersson** [comment]
Ive added the release to banking. I have not tried it with repairing, ill see how that goes

File: common.lic (hunk 1 above)
Comment (jandersson) [inline common.lic]: Is the behaviour undefined?

Comment (rcuhljr) [inline common.lic]: It's very defined, on a fresh boot it's going to deadlock hard. Common can't load waiting on common-arcana to load, common-arcana can't load waiting on common to load.

**Comment by jandersson** [comment]
I could do a lamda with the release code

Sent with AquaMail for Android
http://www.aqua-mail.com

File: common.lic (hunk 1 above)
Comment (jandersson) [inline common.lic]: OK, I'm thinking of copying and pasting in the function body for release_invisibility in the `when` body

File: common.lic [hunk 2]
```diff
@@ -105,7 +103,13 @@ module DRC
 
     case result
     when 'You feel about'
-      release_invisibility
+      # Reduntant copy of DRCA.release_invisibility to avoid deadlocking from including commona-arcana
```
Comment (jandersson) [inline common.lic]: Don't mean redundant in a negative sense btw, just saying DRY is too rigid in this case :]

**Comment by jandersson** [comment]
You cannot give items to the repair guy when invisible so added the release for that. I took out release EOTB like from combat-trainer because this should still cover #945

File: common.lic [hunk 3]
```diff
@@ -105,7 +103,13 @@ module DRC
...
+      get_data('spells')
+        .spell_data
+        .select { |name, properties| properties['invisibility'] }
+        .select { |name, properties| DRSpells.active_spells.keys.include?(name) }
```
Comment (rcuhljr) [inline common.lic]: Yes.

**Comment by rpherbig** [comment]
I wonder if it is time to find a better home for the `rummage` and `get_*` (boxes, skins, etc.) methods. They aren't used in very many places... I'm always in favor of moving code out of `common` and into something more specific.

**Comment by jandersson** [comment]
equipmanager? it has state, but rummage and get_* deal with equipment was my thought

**Comment by rcuhljr** [comment]
I probably still would have preferred to move release_invisibility into common. No objections to this PR though.

**Comment by jandersson** [comment]
I can still do that, it makes sense in removing the dupe code.

Sent with AquaMail for Android
http://www.aqua-mail.com

File: common-money.lic [hunk 4]
```diff
@@ -3,7 +3,7 @@
   Documentation: https://elanthipedia.play.net/Lich_script_development#common-money
 =end
 
-custom_require.call(%w(common common-travel))
+custom_require.call(%w(common common-travel common-arcana))
```
Comment (rpherbig) [inline common-money.lic]: This no longer needs `common-arcana`

File: crossing-repair.lic [hunk 5]
```diff
@@ -2,10 +2,11 @@
   Documentation: https://elanthipedia.play.net/Lich_script_repository#crossing-repair
 =end
 
-custom_require.call(%w(common common-money common-travel drinfomon equipmanager))
+custom_require.call(%w(common common-arcana common-money common-travel drinfomon equipmanager))
```
Comment (rpherbig) [inline crossing-repair.lic]: This no longer needs `common-arcana`

File: crossing-repair.lic [hunk 6]
```diff
@@ -2,10 +2,11 @@
...
 
 class CrossingRepair
   include DRC
+  include DRCA
```
Comment (rpherbig) [inline crossing-repair.lic]: This line can be removed

**Pull request merged**